core/undo_redo.cpp

```diff
@@ -108,7 +108,7 @@ void UndoRedo::add_do_method(Object *p_object, const String &p_method, VARIANT_A
 	Operation do_op;
 	do_op.object = p_object->get_instance_id();
 	if (Object::cast_to<Reference>(p_object)) {
-		do_op.ref = Ref<Reference>(Object::cast_to<Reference>(p_object));
+		do_op.ref = Ref<Reference>(p_object);
```

Comment:
These are just something I noticed during the PR, none of these `cast_tos` are needed as the check has already been doing above.

Reply:
Why not use `is_reference` in the ifs above then?

Reply:
Ah you are correct, `p_object` is already NULL checked, will change. I must have thought it was getting the instance_id and could be NULL. :+1: